### PR TITLE
Update kotlin and libraries to 1.3.0

### PR DESCRIPTION
**Pull request by NikkyAI**
## Description

updated dependencies:

dependency | version
---|---
kotlin | 1.3.0
coroutines | 1.0.0
kotlinx-serialization | 0.9.0

## Type of change

Check all that apply

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (a change which changes the current internal or external interface)
- [X] This change requires a documentation update

## How Has This Been Tested?

In case you did not include tests describe why you and how you have verified the
changes, with instructions so we can reproduce. If you have added comprehensive
tests for your changes, you may omit this section.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new compiler warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Inspect the bytecode viewer, including reasoning why

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -19,7 +19,6 @@ allprojects {
         google()
         mavenCentral()
         jcenter()
-        maven(url = "https://dl.bintray.com/kotlin/kotlin-dev")
```
Comment (SleeplessByte) [inline build.gradle.kts]: If you keep this you don't need to add it again when we want to try out RC

**Comment by iNoles** [comment]
My biggest pet peeve is ImplicitReflectionSerializer that would force third party apps developer to add that code in.

File: build.gradle.kts (hunk 1 above)
Comment (NikkyAI) [reply build.gradle.kts]: i feel lik it makes the intent to use stable 1.3.0 clear and readding it is not that much overhead when we have to also change version numbers and ´plugin stuff in settings.gradle.kts
also it makes sure i did not accidentally pull in dependencies from 1.3.0-rc with mismatching coroutine libraries

**Comment by NikkyAI** [comment]
leaving it out produces compiler errors where it has to be, this is a decision made by jetbrains

File: build.gradle.kts (hunk 1 above)
Comment (SleeplessByte) [reply build.gradle.kts]: Agreed

File: README.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ The easiest HTTP networking library for Kotlin/Android.
 - [x] Special test mode for easier testing
 - [x] Support for reactive programming via RxJava 2.x and Project Reactor 3.x
 - [x] Google Components [LiveData](https://developer.android.com/topic/libraries/architecture/livedata.html) support
-- [x] Built-in object serialization module (Gson, Jackson, Moshi, Forge) :sparkles:
+- [x] Built-in object serialization module (kotlinx-seriualization, Gson, Jackson, Moshi, Forge) :sparkles:
 - [x] Support Kotlin's [Coroutines](https://github.com/Kotlin/kotlinx.coroutines) module
 - [x] API Routing
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply README.md]: Spelling error in `kotlinx-seriualization`

[the remark this answers is not in the document]
Comment (NikkyAI) [reply README.md]: oops

[the remark this answers is not in the document]
Comment (kittinunf) [reply README.md]: I would love to merge everything breaking change at once and do 2.0? Does it make sense?

File: fuel-kotlinx-serialization/src/main/kotlin/com/github/kittinunf/fuel/serialization/FuelSerialization.kt [hunk 3]
```diff
@@ -9,28 +9,33 @@ import com.github.kittinunf.fuel.core.response
...
-import kotlinx.serialization.KSerialLoader
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.ImplicitReflectionSerializer
 import kotlinx.serialization.json.JSON
 import kotlinx.serialization.serializer
 
+@ImplicitReflectionSerializer
 inline fun <reified T : Any> Request.responseObject(
-    loader: KSerialLoader<T> = T::class.serializer(),
+    loader: DeserializationStrategy<T> = T::class.serializer(),
```
Comment (kittinunf) [inline fuel-kotlinx-serialization/src/main/kotlin/com/github/kittinunf/fuel/serialization/FuelSerialization.kt]: This might conflict with #485 though.

**Comment by kittinunf** [comment]
Looks good to me. Shall we merge the 485 in first though?o

File: fuel-kotlinx-serialization/src/main/kotlin/com/github/kittinunf/fuel/serialization/FuelSerialization.kt (hunk 3 above)
Comment (SleeplessByte) [reply fuel-kotlinx-serialization/src/main/kotlin/com/github/kittinunf/fuel/serialization/FuelSerialization.kt]: I made the same change I think. Should merge fine.

**Comment by NikkyAI** [comment]
either way is fine to me, i think adapting this to merge after 485 is barely any effort, so whatever feels right

**Comment by SleeplessByte** [comment]
Let's wait for ✔️ and then merge.

**Pull request merged**